internal/db/fetcher/deleted.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package fetcher
+
+import (
+	"errors"
+
+	"github.com/sourcenetwork/immutable"
+)
+
+// deleted is a fetcher that orchastrates the fetching of deleted and active documents.
+type deleted struct {
```

Comment:
suggestion: Why not rename to `deletedFetcher`, I know the package name is `fetcher` but still prefer the more descriptive name specially as you also have `newDeletedFetcher` function.
```suggestion
// deleted is a fetcher that orchastrates the fetching of deleted and active documents.
type deletedFetcher struct {
```

Reply:
I dislike `fetcher.deletedFetcher` now, and I would have expected @fredcarle to complain if I had named it so.  Happy to rename if the majority want to though, otherwise leaving as-is.

Reply:
Fred and I have rather similar views on this kind of naming stuff and although I'm not speaking for him, I do prefer the explicitness of `deletedFetcher` here. `deleted` is just too generic of a term, same with `document` which given the context of the package could just as easily refer to an actual document.

As for the import/package full naming (ie: `fetcher.deletedFetcher`) this is also OK and not against any language idioms or best practises, because the alternative (`fetcher.deleted`) is not a clear enough naming scheme (ignoring the fact that this isnt a public type).

Reply:
> I dislike `fetcher.deletedFetcher` now, and I would have expected @fredcarle to complain if I had named it so. Happy to rename if the majority want to though, otherwise leaving as-is.

Why `newDeletedFetcher` over `newDeleted` then also haha? The only reason `deleted` is easy to understand for us is because we have context. A new developer just stumbling on this file even though the package is called `fetcher` might not grasp what a general term like `deleted` is referring to here, let alone what a "deleted fetcher" is.

Reply:
My preference after thinking about it over the weekend.
```
mainFetcher or docFetcher
deletedFetcher
versionedFetcher
filteredFetcher
permissionedFetcher
indexedFetcher
```

Reply:
> I don't like both of deleted and deletedFetcher. The documentation for the struct is not deletion-specific, it's for both deleted and active. If the main thing this fetcher does is orchestrating something, let's call it orchestratingFetcher.

I can remove references to the deleted-ness of that type.  It really just manages the fetching of documents from multiple statuses, nothing there is delete specific.  Abstracting the naming will make the code a little more obscure IMO, but it is probably more correct.

New naming for it will probably be `statusFetcher` or similar.

If anyone objects to this change please let me know :)

- [ ] Remove deleted-ness of `deleted` fetcher

Reply:
> mainFetcher or docFetcher

:) I don't see myself being able to call it `mainFetcher`, and having a `docFetcher` *and* a `documentFetcher` in the same code-space is quite unpleasant IMO.  I might tweak the `wrapper` to `wrappingFetcher` or `bridgeFetcher`/`tempFetcher` if no better names pop up in the meantime.

- [ ] Rename `wrapper

Reply:
> `Fetcher` suffix

People are in favour of this, so I'll add the suffix to all.

- [ ] Add `Fetcher` suffix to type names.

Reply:
> `docFetcher` == `documentFetcher`. There is no duplication intended :)

Ah! I thought that was a rename of `wrapper` (because of it being at the top, and the `main` suggestion) :grin: